blocks/library/image/index.js

```diff
@@ -72,6 +75,23 @@ registerBlock( 'core/image', {
 	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, alt, caption } = attributes;
 
+		if ( ! url ) {
+			return (
```

Comment:
>It means when it acts as a placeholder, when no content is yet set. This is going to be a prevalent case for templates down the road.

Right, but in the sense of treating blocks as rendering always from its state, an imperative "renderNew" doesn't make sense. A block could implement conditions under which it could be considered empty (here, `! url`), but it's unclear how we'd want to generalize this. Maybe related to #609, an attribute can be flagged as `required` and, when missing, use a separate render function like your `empty` example.

I don't know that introducing this break from `edit` / `save` duality really provide a whole lot of value though for what could be equally represented by:

```js
edit: ( { attributes } ) =>
	attributes.url
		? <ImageEdit />
		: <EmptyImageEdit />
```